### PR TITLE
update motivation section

### PR DESCRIPTION
**Pull request by MylesBorins**
this incorporates slides from @domenic and restructures current content

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,31 +12,51 @@ Since the decision to delay standardizing top-level `await` it has come up in a
 
 ## Motivation
 
-The current implementation of `async / await` only support the `await` keyword inside of `async` functions. As such we are beginning to see the following pattern emerge:
+### top-level main
```
Comment (domenic) [inline README.md]: I think top-level main and IIAFEs are basically the same thing. Maybe fine to have as separate examples, but worth being aware.

File: README.md [hunk 2]
```diff
@@ -12,31 +12,51 @@ Since the decision to delay standardizing top-level `await` it has come up in a
...
 export ...
 ```
 
-This pattern is reminiscent of the classic pattern of wrapping all of your code in a self executing function `(fn(){}())`. This type of "magic" does not benefit new developers and creates excessive repeated code throughout the ecosystem.
+This pattern is creating an abundance of boilerplate code in the ecosystem. If the pattern is repeated throughout the module graph it degrages the determinism of the execution.
 
-Another risk with the above pattern is that it makes the body of the function asynchronous. If people are utilizing this pattern throughout their graph there will no longer be a deterministic execution order.
+### IIAFEs
 
-Another pattern that is beginning to surface is exporting async function and awaiting the results of imports, which drastically impacts our ability to do static analysis of the module graph.
```
Comment (domenic) [inline README.md]: This paragraph is good and could move into the completely dynamic modules section.

File: README.md (hunk 1 above)
Comment (MylesBorins) [reply README.md]: Indeed. I'm seeing the top-level main being used more than IIAFEs at the moment, so I was trying to illustrate that one pattern will eventually lead to the other. Perhaps that is unnecessary

File: README.md [hunk 3]
```diff
@@ -12,31 +12,51 @@ Since the decision to delay standardizing top-level `await` it has come up in a
...
+  // main program goes here...
+})();
+```
+
+###  completely dynamic modules
+
+```mjs
+export default async () => {
+  // import other modules like this one
+  const import1 = await (await import('./import1.mjs'))();
```
Comment (bmeck) [inline README.md]: missing `.default`

File: README.md (hunk 1 above)
Comment (domenic) [reply README.md]: Ah OK. Might make sense to add some glue text.

File: README.md (hunk 3 above)
Comment (MylesBorins) [reply README.md]: can you share the code example you would rather see?

Comment (bmeck) [reply README.md]: `import` returns the full namespaces so the above ~is:

```mjs
const import1Namespace = await import('./import1.mjs');
const import1 = await import1Namespace;
```

So it doesn't actually call / use the `export default` of `import1.mjs`.

```mjs
const import1 = await ((await import('./import1.mjs')).default)();
```

looks like what was intended.

**Pull request merged**